packages/twenty-front/src/modules/workflow/workflow-trigger/components/WorkflowEditTriggerCronForm.tsx

```diff
@@ -0,0 +1,305 @@
+import { WorkflowCronTrigger } from '@/workflow/types/Workflow';
+import { useIcons } from 'twenty-ui';
+import { useTheme } from '@emotion/react';
+import { Select } from '@/ui/input/components/Select';
+import { WorkflowStepHeader } from '@/workflow/workflow-steps/components/WorkflowStepHeader';
+import { WorkflowStepBody } from '@/workflow/workflow-steps/components/WorkflowStepBody';
+import { getTriggerIcon } from '@/workflow/workflow-trigger/utils/getTriggerIcon';
+import { isDefined } from 'twenty-shared';
+import { getTriggerDefaultLabel } from '@/workflow/workflow-trigger/utils/getTriggerLabel';
+import { CRON_TRIGGER_INTERVAL_OPTIONS } from '@/workflow/workflow-trigger/constants/CronTriggerIntervalOptions';
+import cron from 'cron-validate';
+import { useState } from 'react';
+import { FormTextFieldInput } from '@/object-record/record-field/form-types/components/FormTextFieldInput';
+import { FormNumberFieldInput } from '@/object-record/record-field/form-types/components/FormNumberFieldInput';
+import { isNumber } from '@sniptt/guards';
+import { getCronTriggerDefaultSettings } from '@/workflow/workflow-trigger/utils/getCronTriggerDefaultSettings';
+
+type WorkflowEditTriggerCronFormProps = {
+  trigger: WorkflowCronTrigger;
+  triggerOptions:
+    | {
+        readonly: true;
+        onTriggerUpdate?: undefined;
+      }
+    | {
+        readonly?: false;
+        onTriggerUpdate: (trigger: WorkflowCronTrigger) => void;
+      };
+};
+
+export const WorkflowEditTriggerCronForm = ({
+  trigger,
+  triggerOptions,
+}: WorkflowEditTriggerCronFormProps) => {
+  const theme = useTheme();
+  const [errorMessages, setErrorMessages] = useState<{
+    [key: number]: string | undefined;
+  }>({});
+
+  const { getIcon } = useIcons();
+
+  const headerIcon = getTriggerIcon({
+    type: 'CRON',
+  });
+
+  const defaultLabel =
+    getTriggerDefaultLabel({
+      type: 'CRON',
+    }) ?? '';
+
+  const headerTitle = isDefined(trigger.name) ? trigger.name : defaultLabel;
+
+  const headerType = 'Trigger';
+
+  return (
+    <>
+      <WorkflowStepHeader
+        onTitleChange={(newName: string) => {
+          if (triggerOptions.readonly === true) {
+            return;
+          }
+
+          triggerOptions.onTriggerUpdate({
+            ...trigger,
+            name: newName,
+          });
+        }}
+        Icon={getIcon(headerIcon)}
+        iconColor={theme.font.color.tertiary}
+        initialTitle={headerTitle}
+        headerType={headerType}
+      />
+      <WorkflowStepBody>
+        <Select
+          dropdownId="workflow-edit-cron-trigger-interval"
+          label="Trigger interval"
+          fullWidth
+          disabled={triggerOptions.readonly}
+          value={trigger.settings.type}
+          options={CRON_TRIGGER_INTERVAL_OPTIONS}
+          onChange={(newTriggerType) => {
```

Comment:
I think the validation errors should be reset when changing the trigger interval. The values of the inputs are reset, the validation errors should be too.


https://github.com/user-attachments/assets/0e2cad6a-6734-420c-b9d8-864a60fc0b15